tsdb/querier.go

```diff
@@ -336,9 +337,9 @@ func PostingsForMatchers(ix IndexReader, ms ...labels.Matcher) (index.Postings,
 	}
 
 	for _, m := range ms {
-		matchesEmpty := m.Matches("")
-		if labelMustBeSet[m.Name()] || !matchesEmpty {
+		if labelMustBeSet[m.Name()] {
```

Comment:
You're changing the logic here, which will be slower

Reply:
If `labelMustBeSet[m.Name()]` is true, then `!matchesEmpty` is not need to judge.

If `labelMustBeSet[m.Name()]` is false, then `!matchesEmpty` must be false.

The judgment of `!matchesEmpty` is redundant, so the refactor is fine and make the code more clear. WDYT?

Reply:
That makes sense.